### DISCUSSION TITLE
IPC notification mechanism, prefer UDS over semaphore on SHM?

### DISCUSSION THREAD

**Discussion by IIIIIIIIll** [eclipse-iceoryx/iceoryx2#551] [Q&A] [answered]
First of all thank you for the great work! I got ~2μs latency with ipc event on a 14900k which is super impressive. 

However I do have a question when I looked into the implementation details. I noticed as of version 0.4.1, iox2 uses unix domain socket for ipc event notification but there exists a `event::sem_bitset_posix_shared_memory::Event` in the codebase. After some digging in the repo., I realized the code was switched from using semaphore on SHM to UDS (commit# 5473ee15). Is this done by some particular consideration? As I would imagine semaphore on SHM should be better in terms of latency and CPU consumption.  Thank you!

**Comment by elBoberido** [comment, marked as answer]
@IIIIIIIIll the main reason for the switch was that we need to support file descriptors in the `WaitSet`. With semaphores this is unfortunately not possible. But the performance impact should be negligible in real word scenarios, since the OS will most probably need way more than 2µs to wake up a process which is not active.

**Reply by IIIIIIIIll** [reply]
Thank you for your reply! I have yet to read the `Waitset` part but I will surely take a look.
